server/xpub-controller/helpers/files.js

```diff
@@ -58,6 +58,26 @@ class FilesHelper {
     throw new Error('Invalid parameters to calculate the upload file progress.')
   }
 
+  static startFileProgress(
+    pubsub,
+    ON_UPLOAD_PROGRESS,
+    startedTime,
+    predictedTime,
+    manuscriptId,
+    interval,
+  ) {
+    return setInterval(
```

Comment:
Added this function to let me mock the calling of `setInterval` in the  maniuscript upload controller unit test without using jest.useFakeTimers as this was throwing errors. see #1553 for a little more context.